### ISSUE TITLE
How often is the validation/evaluation performed using run_classifier.py?

### ISSUE THREAD

**Issue by justinas-kazanavicius** [google-research/bert#636]
I have noticed that evaluation on the dev set is not being regularly performed on my task, not even after every epoch, which made my model overfit over time. One time the evaluation on the dev set did work after finishing 3 epochs when `num_epochs` to 3. However, it has never happened when I've set it to 30 which it has never reached. Does this mean it only performs the dev evaluation at the end of all epochs? Or is it because my dev set might be too large?

Could it be because I was using my University's supercomputers where I had to re-submit training jobs every 48 hours from the last checkpoint? Maybe this caused the evaluation step to never be reached?

If anyone knows the logic of evaluation using `run_classifier.py` that would be very helpful.

Thank you.

**Comment by guotong1988** [comment]
Same question.

**Comment by plkumjorn** [comment]
I think it would be great if we could evaluate for every training loop and stop training if the evaluation loss doesn't improve for n evaluations (i.e., early stopping).

**Comment by AakritiBudhraja** [comment]
Any information on this? I tried to implement it (for fine tuning in run_classifier.py) just like the way it has been done in transformer. But it looks like the global_step is getting stuck after 1 epoch of training.
